src/Http/Routing/src/ValidationEndpointFilterFactory.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Http.Validation;
+
+internal static class ValidationEndpointFilterFactory
+{
+    private const string ValidationContextJustification = "The DisplayName property is always statically initialized in the ValidationContext through this codepath.";
+
+    [UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code", Justification = ValidationContextJustification)]
```

Comment:
Can you minimize where this suppression applies? We shouldn't suppress on whole methods so more warnings don't sneak in.

Reply:
> Can you minimize where this suppression applies? We shouldn't suppress on whole methods so more warnings don't sneak in.

Good call. Done.

> Also, I don't see where DisplayName is being set.

It was being set in the `Validate` of the `ValidatableParameterInfo`. I think that's still valid to do since it's set before the `getter` is ever invoked but for safe measure I added an explicit set in the endpoint filter as well.